compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -16,18 +31,23 @@ pub fn is_const_evaluatable<'cx, 'tcx>(
 ) -> Result<(), ErrorHandled> {
     debug!("is_const_evaluatable({:?}, {:?})", def, substs);
     if infcx.tcx.features().const_evaluatable_checked {
```

Comment:
If I'm reading the code right, this is not needed as `AbstractConst::new` will return `None` if the feature gate is not enabled

Reply:
I kept it in as we otherwise have to reason globally here.

We might also use const evaluatability if we check for constants from extern crates if they use this feature without this,
which I don't actually want. (so that we can use it in `libs` without worrying about that)